src/backend/cod1.c

```diff
@@ -4515,7 +4525,16 @@ code *loaddata(elem *e,regm_t *pretregs)
     }
     else if (sz <= REGSIZE)
     {
-        ce = loadea(e,&cs,0x8B,reg,0,RMload,0); // MOV reg,data
+        unsigned op = 0x8B;                     // MOV reg,data
+        if (sz == 2 && !I16 && config.target_cpu >= TARGET_PentiumPro &&
+            // Workaround for OSX linker bug:
+            //   ld: GOT load reloc does not point to a movq instruction in test42 for x86_64
+            !(config.exe & EX_OSX64 && !(sytab[e->EV.sp.Vsym->Sclass] & SCSS))
+           )
+        {
+            op = tyuns(tym) ? 0x0FB7 : 0x0FBF;  // MOVZX/MOVSX
```

Comment:
The trouble with that is most of the instruction mnemonics have multiple encodings. If I invent my own mnemonics, it gets hard to compare the code generator code side by side with the Intel CPU manuals.

Hardcoded numbers also make comparing the object file dumps with the code generator code easier without having to go through another translation layer.